src/Parallel/Main.hpp

```diff
@@ -601,12 +584,40 @@ void Main<Metavariables>::
       tmpl::filter<component_list,
                    Parallel::is_array_proxy<tmpl::bind<
                        Parallel::proxy_from_parallel_component, tmpl::_1>>>;
-  tmpl::for_each<array_component_list>([this](auto parallel_component_v) {
+  // Put each singleton on a different proc. In the future, this should be
+  // changed so that no DG elements get placed on these procs so singletons have
+  // thier own procs.
```

Comment:
What are your plans for distributing singletons to cores? Do you think we should allow each singleton component to choose the core by itself, or perhaps have the metavars of every executable decide which set of cores is available for each parallel component? I'm thinking we may have a number of cheap singletons that are absolutely fine sharing PE 0 with DG elements, and blocking a PE for each singleton may not be the best. For example, the elliptic solver uses a bunch of "residual monitor" singletons that are very cheap and don't need to run on their own private core. Also, I expect interpolation targets (which are singleton components) often don't need to run on their own private core, except of course of expensive things like the AH finder.

Reply:
I'm currently trying to come up with a scheme to best distribute everything. However, this will involve modifying where we put DG elements because they can no longer be on all cores if we are reserving some. So this will be a separate PR. For now I settled to just put each singleton on its own core where DG elements can also occupy those cores.

I agree that probably not all singletons need their own cores. The control system singletons themselves are also fairly cheap so most won't need their own core (except maybe size control). I'll take this into account. Maybe the singletons themselves have a `constexpr bool` saying they need their own core?

Reply:
Yea let's decide these things in a follow-up PR, the round-robin distribution you chose here for now is good.

When coming up with a scheme, please take into account that we'll probably need to experiment with it, so don't make everything compile-time choices. Once things are implemented at compile-time then it's hard to switch back to runtime choices. Prefer input-file options where a compile-time choice doesn't give any benefits.